src/serac/physics/utilities/finite_element_vector.hpp

```diff
@@ -0,0 +1,218 @@
+
+  /**
+   * @brief Set a finite element state to a constant value
+   *
+   * @param value The constant to set the finite element state to
+   * @return The modified finite element state
+   * @note This sets the true degrees of freedom and then broadcasts to the shared grid function entries. This means
+   * that if a different value is given on different processors, a shared DOF will be set to the owning processor value.
+   */
+  FiniteElementVector& operator=(const double value);
+
+  /**
+   * @brief Utility function for creating a tensor, e.g. mfem::HypreParVector,
+   * mfem::ParBilinearForm, etc on the FESpace encapsulated by an FEState object
+   * @return An owning pointer to a heap-allocated tensor
+   * @pre Tensor must have the constructor Tensor::Tensor(ParFiniteElementSpace*)
+   */
+  template <typename Tensor>
+  std::unique_ptr<Tensor> createOnSpace()
+  {
+    static_assert(std::is_constructible_v<Tensor, mfem::ParFiniteElementSpace*>,
+                  "Tensor must be constructible with a ptr to ParFESpace");
+    return std::make_unique<Tensor>(&detail::retrieve(space_));
+  }
+
+protected:
+  /**
+   * @brief A reference to the mesh object on which the field is defined
+   */
+  std::reference_wrapper<mfem::ParMesh> mesh_;
+  /**
+   * @brief Possibly-owning handle to the FiniteElementCollection, as it is owned
+   * by the FiniteElementVector in a normal run and by the MFEMSidreDataCollection
+   * in a restart run
+   * @note Must be const as FESpaces store a const reference to their FEColls
+   */
+  detail::MaybeOwningPointer<const mfem::FiniteElementCollection> coll_;
+  /**
+   * @brief Possibly-owning handle to the FiniteElementCollection, as it is owned
+   * by the FiniteElementVector in a normal run and by the MFEMSidreDataCollection
+   * in a restart run
+   */
+  detail::MaybeOwningPointer<mfem::ParFiniteElementSpace> space_;
+  /**
+   * @brief Possibly-owning handle to the ParGridFunction, as it is owned
+   * by the FiniteElementVector in a normal run and by the MFEMSidreDataCollection
+   * in a restart run
+   */
+  detail::MaybeOwningPointer<mfem::ParGridFunction> gf_;
+  mfem::HypreParVector                              true_vec_;
+  std::string                                       name_ = "";
+};
+
+/**
+ * @brief Find the average value of a finite element vector across all dofs
+ *
+ * @param state The state variable to compute a max of
+ * @return The average value
+ */
+double avg(const FiniteElementVector& fe_vector);
+
+/**
+ * @brief Find the max value of a finite element vector across all dofs
+ *
+ * @param state The state variable to compute a max of
+ * @return The max value
+ */
+double max(const FiniteElementVector& fe_vector);
+
+/**
+ * @brief Find the min value of a finite element vector across all dofs
+ *
+ * @param state The state variable to compute a min of
+ * @return The min value
+ */
+double min(const FiniteElementVector& fe_vector);
```

Comment:
It might be worth adding that these operations act directly on the components of the field, rather than the actual field itself (e.g. the maximum displacement in a domain may not occur at a node)